Enhancements/additions to kctrl CLI

100mik opened the pull request:
Ongoing PR with progress on #412 

So far:
- Flag for fetching kapp-controller version on the server
- StdIn input while passing a values file

100mik committed:
Add flag to view controller version. TODO: Give more thought to how we wanna deal with kapp-controller installed in non-default namespaces

100mik committed:
Support StdIn input while adding values file

100mik committed:
Rephrasing 'kapp controller' => 'kapp-controller' while displaying controller version

cppforlife commented on cli/pkg/kctrl/cmd/version.go:
```diff
@@ -26,11 +38,42 @@ func NewVersionCmd(o *VersionOptions, flagsFactory cmdcore.FlagsFactory) *cobra.
 func (o *VersionOptions) Run() error {
+	if o.controllerVersion {
+		err := o.showControllerVersion()
+		if err != nil {
+			return err
+		}
```
nit: just return above

praveenrewar replied:
Done

cppforlife commented on cli/pkg/kctrl/cmd/version.go:
```diff
@@ -26,11 +38,42 @@ func NewVersionCmd(o *VersionOptions, flagsFactory cmdcore.FlagsFactory) *cobra.
+	controllerDeployment, err := coreClient.AppsV1().Deployments(kappControllerNamespace).Get(context.Background(), kappControllerDeployment, metav1.GetOptions{})
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return fmt.Errorf("kapp-controller not installed on cluster")
```
nit: `kapp-controller not installed on cluster (or is not present in the default location namespace=kapp-controller deployment=kapp-controller)`

praveenrewar replied:
Done

praveenrewar committed:
Use structured examples in package available get

praveenrewar committed:
Use structured examples in package installed

praveenrewar committed:
Use structured examples in package repository

praveenrewar committed:
Make example asString private

praveenrewar committed:
Update example usage to be inline

praveenrewar committed:
Update copyright information

praveenrewar committed:
Use PackageCommandTreeOpts to set binaryName

Use PackageCommandTreeOpts to set binaryName and positionalNameArgs and enable or disable color or json flags

cppforlife commented on cli/pkg/kctrl/cmd/package/available/get.go:
```diff
@@ -30,11 +30,12 @@ type GetOptions struct {
 
 	ValuesSchema bool
 
+	binaryName        string
 	positionalNameArg bool
```
should we turn these into a struct (it's used all over) and it gets pretty noisy to pass them around as individual args.

praveenrewar replied:
Yep, I was thinking of doing the same, I was a little hesitant because then we would have opts inside other opts (GetOptions, ListOptions, etc,.), but I think it makes sense to have a struct, maybe we can name it something else.

praveenrewar committed:
Update error message for kctrl version

praveenrewar committed:
Use cmd.Args for positionalArgs

praveenrewar committed:
Update command usage for positional arguments

praveenrewar committed:
Add silence usage to sub commands

praveenrewar committed:
Add annotations to get and list commands

praveenrewar committed:
Move package command tree opts to cmdcore
